gprofiler/profilers/profiler_base.py

```diff
@@ -196,65 +185,79 @@ class SpawningProcessProfilerBase(ProcessProfilerBase):
+        self._profile_spawned_processes = profile_spawned_processes
         self._submit_lock = Lock()
         self._threads: Optional[ThreadPoolExecutor] = None
         self._start_ts: Optional[float] = None
+        self._preexisting_pids: Optional[List[int]] = None
         self._enabled_proc_events = False
-        self._futures: Dict[Future, int] = {}
+        self._futures: Dict[Future, Tuple[int, str]] = {}
         self._sched = sched.scheduler()
         self._sched_stop = False
         self._sched_thread = Thread(target=self._sched_thread_run)
 
-    def _should_profile_process(self, pid: int) -> bool:
+    def _should_profile_process(self, process: Process) -> bool:
         raise NotImplementedError
 
     def _notify_selected_processes(self, processes: List[Process]) -> None:
-        # TODO ensure PIDs in _proc_exec_callback don't intersect with "processes"?
         # now we start watching for new processes.
-        self._start_profiling_spawning()
+        self._start_profiling_spawning(processes)
 
     @property
     def _is_profiling_spawning(self) -> bool:
         return self._threads is not None
 
-    def _start_profiling_spawning(self) -> None:
+    def _start_profiling_spawning(self, processes: List[Process]) -> None:
+        logger.debug(f"{self.__class__.__name__}: starting profiling spawning processes")
         with self._submit_lock:
             self._start_ts = time.monotonic()
             self._threads = ThreadPoolExecutor()
+            self._preexisting_pids = list(map(lambda p: p.pid, processes))
 
     def _stop_profiling_spawning(self) -> None:
         with self._submit_lock:
             self._start_ts = None
             assert self._threads is not None
             self._threads.shutdown()  # waits (although - all are done by now)
             self._threads = None
+            self._preexisting_pids = None
 
     def _proc_exec_callback(self, tid: int, pid: int) -> None:
-        self._sched.enter(self._BACKOFF_INIT, 0, self._check_process, (Process(pid), self._BACKOFF_INIT))
+        with contextlib.suppress(NoSuchProcess):
+            self._sched.enter(self._BACKOFF_INIT, 0, self._check_process, (Process(pid), self._BACKOFF_INIT))
 
     def start(self) -> None:
         super().start()
 
-        self._sched_thread.start()
+        if self._profile_spawned_processes:
+            self._sched_thread.start()
 
-        try:
-            register_exec_callback(self._proc_exec_callback)
-        except Exception:
-            logger.warning("Failed to enable proc_events listener for executed processes", exc_info=True)
-        else:
-            self._enabled_proc_events = True
+            try:
+                register_exec_callback(self._proc_exec_callback)
+            except Exception:
+                logger.warning("Failed to enable proc_events listener for executed processes", exc_info=True)
+            else:
+                self._enabled_proc_events = True
```

Comment:
Let's name `_enabled_proc_events` differently so there's no chance `java.py` could accidentally use it. (`_enabled_proc_events_spawning`? idk)